app/src/main/java/com/hedvig/app/ui/compose/theme/Colors.kt

```diff
@@ -10,13 +15,60 @@ val background = Color(0xffF6F6F6)
 
 val whiteHighEmphasis = Color(0xFFFAFAFA)
 
-val lavender300 = Color(0xffC9ABF5)
-val lavender400 = Color(0xffBE9BF3)
-
 val errorLight = Color(0xffDD2727)
 val errorDark = Color(0xffE24646)
 
 val textColorPrimary = Color(0xAB121212)
 val textColorPrimaryDark = Color(0x8FFAFAFA)
 
 val surfaceDark = Color(0xffBE9BF3)
+
+val lavender200: Color
+    @Composable get() = colorResource(R.color.lavender_200)
+val lavender300: Color
+    @Composable get() = colorResource(R.color.lavender_300)
+val lavender400: Color
+    @Composable get() = colorResource(R.color.lavender_400)
+val lavender900: Color
+    @Composable get() = colorResource(R.color.lavender_900)
+
+val foreverOrange300: Color
+    @Composable get() = colorResource(R.color.forever_orange_300)
+val foreverOrange500: Color
+    @Composable get() = colorResource(R.color.forever_orange_500)
+
+object ClaimStatusColors {
+    object Pill {
+        val paid: Color
+            @Composable get() = if (isSystemInDarkTheme()) {
+                lavender400
+            } else {
+                lavender200
+            }
+
+        val reopened: Color
+            @Composable get() = if (isSystemInDarkTheme()) {
+                foreverOrange500
+            } else {
+                foreverOrange300
+            }
+    }
+
+    object Progress {
+        val paid: Color
+            @Composable get() = colorResource(R.color.colorSecondary)
+
+        val reopened: Color
+            @Composable get() = foreverOrange500
+    }
+}
+
+@Composable
+fun hedvigContentColorFor(backgroundColor: Color): Color {
+    // TODO check how to properly provide contentColor colors for non-material colors. This is a workaround for now.
+    return when (backgroundColor) {
+        lavender400 -> Color.Black
+        foreverOrange500 -> Color.Black
+        else -> contentColorFor(backgroundColor)
+    }
+}
```

Comment:
Context: Text on custom colors is off in dark mode without this.
So this is something that I am not familiar with.
Using contentColorFor() works great but *just* when using the normal material colors, since under the hood all it does is check if the color is primary, it uses onPrimary for the content, if it's secondary it uses onSecondary etc. But here we have unique colors that aren't part of the material spec, just some unique color for the pills. This workaround works for now, but is there a better way to do this? Is there a function that can infer from a color's luminescence or whatever if the content color should be white or black?

Reply:
I'm not quite sure about the particular use-case for this function, but this is probably good enough, I guess? 😄